test/CMakeLists.txt

```diff
@@ -1,30 +1,63 @@
 # only build tests on targets that declare they are like posix
 if(DEFINED TARGET_LIKE_X86_WINDOWS_NATIVE OR DEFINED TARGET_LIKE_X86_LINUX_NATIVE OR DEFINED TARGET_LIKE_X86_OSX_NATIVE)
 
-    # describe the test executable
-    add_executable(mbed_client_cli_test Test.cpp)
-
     include_directories("../yotta_modules/cpputest")
-
-    # describe what the test executable needs to link with
-    target_link_libraries(mbed_client_cli_test
-        "mbed-client-cli"
-        "mbed-trace"
-        cpputest
-    )
     add_definitions("-Wno-write-strings")
 
-    # describe what is actual test binary
-    if(DEFINED TARGET_LIKE_X86_WINDOWS_NATIVE)
-        SET(TEST_EXECUTABLE "build/x86-windows-native/test/mbed_client_cli_test")
-    elseif(DEFINED TARGET_LIKE_X86_OSX_NATIVE)
-         SET(TEST_EXECUTABLE "../../../build/x86-osx-native/test/mbed_client_cli_test")
-         add_test(mbed_client_cli_test ${TEST_EXECUTABLE})
-         add_dependencies(all_tests mbed_client_cli_test)
-    elseif(DEFINED TARGET_LIKE_X86_LINUX_NATIVE)
-        SET(TEST_EXECUTABLE "../../../build/x86-linux-native/test/mbed_client_cli_test")
-    endif()
-
-    add_test(mbed_client_cli_test ${TEST_EXECUTABLE})
+    # FULL BUILD TESTS
+    set(FLAGS
+            MBED_CONF_CMDLINE_USE_MINIMUM_SET=0
+            MBED_CONF_CMDLINE_USE_MINIMUM_SET=1
+            )
+    set(TESTS
+            full
+            min
+            )
+    foreach(flag ${FLAGS})
+        list(FIND FLAGS ${flag} index)
+        list(GET TESTS ${index} TEST)
+
+        MESSAGE("TEST flag: ${flag}")
+
+        add_library( mbed-client-cli-${TEST}
+            ../source/ns_cmdline.c
+            ../source/ns_list_internal/ns_list.c
+            )
+        if(DEFINED TARGET_LIKE_X86_LINUX_NATIVE)
+            if(${TEST} STREQUAL "full")
+                MESSAGE("USE GCOV BUILD")
+                set_target_properties(mbed-client-cli-${TEST} PROPERTIES -fprofile-arcs -ftest-coverage)
```

Comment:
@TeroJaasko @SeppoTakalo any idea why this does not work ? 
How to add those options gcov build ?

Reply:
Syntax is not correct.. I think its `PROPERTIES prop1 val1 prop2 val2`
https://cmake.org/cmake/help/latest/command/set_target_properties.html

I think these needs to go into CFLAGS, CPPFLAGS and probably even LDFLAGS.
https://gcc.gnu.org/onlinedocs/gcc/Invoking-Gcov.html#Invoking-Gcov does not properly specify

Some discussion here:
https://stackoverflow.com/questions/13116488/detailed-guide-on-using-gcov-with-cmake-cdash

Reply:
This might even work:
```
set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} --coverage -fprofile-arcs -ftest-coverage")
set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} --coverage -fprofile-arcs -ftest-coverage")
```

Reply:
its not suitable for now - i want to capture coverage only for ”full” build...🙄

Reply:
this was enough earlier:
```
add_definitions("-g -O0 -fprofile-arcs -ftest-coverage")
target_link_libraries(mbed-client-cli gcov)
```